test/test-files/event-bus.ts

```diff
@@ -11,14 +11,34 @@ const func = new Function<undefined, { source: string }>(
 );
 
 /**
- * Invalid integrations in transformers
+ * Invalid integrations in transformers (minimal payload)
  */
 bus.all().map(() => {
   // eslint-disable-next-line @typescript-eslint/no-floating-promises
   bus.putEvents({ "detail-type": "detail", detail: {}, source: "" });
   return {};
 });
 
+/**
+ * Invalid integrations in transformers (maximal payload)
+ */
+bus.all().map(() => {
+  // eslint-disable-next-line @typescript-eslint/no-floating-promises
+  bus.putEvents({
+    "detail-type": "detail",
+    detail: {},
+    source: "",
+    version: "",
+    id: "",
+    account: "",
+    time: "",
+    region: "",
+    resources: [""],
+    "trace-header": "",
+  });
+  return {};
+});
+
```

Comment:
You don't need to add this here, this just tests our validations and your change doesn't change the validations.

However, could you add your field here:

https://github.com/functionless/functionless/blob/main/test/function.localstack.test.ts#L433

That will test the new field transform in lambda

and here: https://github.com/functionless/functionless/blob/main/test/step-function.test.ts#L818

To test the step function transform?

Reply:
Will do, commit incoming shortly.

Reply:
Do we need to validate that trace header is not used in the event pattern?

Reply:
That's a very good point. If not, it could cause a frustrating developer experience.

Reply:
See my comment above about which type to put it on. If we don't add it to `Event` and instead add it to `PutEventInput`, it shouldn't create confusion.